web/Web/web/CAL/models.py

```diff
@@ -0,0 +1,38 @@
+from config.settings.base import AUTH_USER_MODEL as User
+
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+
+from web.core.models import Session
+
+
+class DS_logging(models.Model):
+    class Meta:
+        unique_together = ['user', 'stratum_num', 'session']
+        index_together = ['user', 'stratum_num', 'session']
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    session = models.ForeignKey(Session, on_delete=models.CASCADE)
+
+    stratum_size = models.IntegerField(default=-1)
```

Comment:
Instead of having a default value, all these fields must not be null. `IntegerField(null=False, blank=False)`

Reply:
Fixed